src/reason_pprint_ast.ml

```diff
@@ -2855,118 +2980,89 @@ class printer  ()= object(self:'self)
             ""
             (self#simple_pattern p)
 
-  (**
-   * TODO: Remove all this special syntax for mutatively setting things.
-   * Save the "monadic" arrow for some other more valuable pattner and
-   * remove a ton of complexity from the parser/printer.
-   *)
-  method sugar_expr e =
-    let access op cls e1 e2 = makeList ~interleaveComments:false [
-      (* Important that this be not breaking - at least to preserve same
-         behavior as stock desugarer. It might even be required (double check
-         in parser.mly) *)
-      e1;
-      atom ".";
-      atom op;
-      e2;
-      atom cls;
-    ] in
+  method access op cls e1 e2 = makeList ~interleaveComments:false [
+    (* Important that this be not breaking - at least to preserve same
+       behavior as stock desugarer. It might even be required (double check
+       in parser.mly) *)
+    e1;
+    atom ".";
+    atom op;
+    e2;
+    atom cls;
+  ]
+
+  method simple_get_application e =
+    match (e.pexp_desc, e.pexp_attributes) with
+    | (_, attrHd::attrTl) -> None (* Has some attributes - not simple *)
+    | (Pexp_apply (eFun, ls), []) -> (
+      match (printedStringAndFixityExpr eFun, ls) with
+      (* We must take care not to print two subsequent prefix operators without
+         spaces between them (`! !` could become `!!` which is totally
+         different).  *)
+      | (AlmostSimplePrefix prefixStr, [("", rightExpr)]) ->
```

Comment:
Is it an invariant that a "AlmostSimplePrefix" always has its argument list as [("", rightExpr)]?